qemu/softmmu_template.h

```diff
@@ -184,17 +184,22 @@ WORD_TYPE helper_le_ld_name(CPUArchState *env, target_ulong addr, int mmu_idx,
 
     // memory might be still unmapped while reading or fetching
     if (mr == NULL) {
+        struct hook *hook;
+        int handled = 0;
 #if defined(SOFTMMU_CODE_ACCESS)
         error_code = UC_ERR_FETCH_UNMAPPED;
-        if (uc->hook_mem_fetch_idx != 0 && ((uc_cb_eventmem_t)uc->hook_callbacks[uc->hook_mem_fetch_idx].callback)(
-                                       uc, UC_MEM_FETCH_UNMAPPED, addr, DATA_SIZE, 0,
-                                       uc->hook_callbacks[uc->hook_mem_fetch_idx].user_data)) {
+        uc_hook_loop_range(uc, hook, UC_HOOK_MEM_FETCH_UNMAPPED, addr) {
+            if ((handled = ((uc_cb_eventmem_t)hook->callback)(uc, UC_MEM_FETCH_UNMAPPED, addr, DATA_SIZE, 0, hook->user_data)))
```

Comment:
indentation here looks correct to me. this is a loop macro to make looping over the hooks not a nightmare. I originally had a visible `for ()` statement involved but I couldn't get it to work with c89.